middle_end/flambda2/simplify/simplify_apply_expr.ml

```diff
@@ -394,11 +403,19 @@ let simplify_direct_partial_application ~simplify_expr dacc apply
       let args =
         List.map arg applied_args @ List.map BP.simple remaining_params
       in
+
+      (* Under-applications are converted in full applications. Let's assume
```

Comment:
Let's be consistent please with "partial applications" not "under applications".  More importantly, this code path is only for full applications, not partial applications -- please check.

Reply:
I disagree, this function is simplifying partial applications. However, it is true that the position of the comment itself (just before the generation of the full application corresponding to this partial application) might make things confusing. 
I've moved the comment to the top of the function

Reply:
Ah sorry, I was misreading the github view.